### PR TITLE
Judicial system/Strict TS setting pt.1

### PR DESCRIPTION
**Pull request by oddsson**
# Start using strict TS.

https://app.asana.com/0/1197628843274291/1199143603683037

## What

See ☝️

## Why

It's a requirement from island-is. Also it's generally good practice.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against master before asking for a review

### CODE REVIEW

**Review by gudjong — APPROVED** [review]

File: apps/judicial-system/web/src/routes/DetentionRequests/DetentionRequests.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const CasesQuery = gql`
 `
 
 export const DetentionRequests: React.FC = () => {
-  const [cases, setCases] = useState<Case[]>(null)
+  const [cases, setCases] = useState<Case[] | null>(null)
```
Comment (addi) [inline apps/judicial-system/web/src/routes/DetentionRequests/DetentionRequests.tsx:45]: Should we not use
`useState<Case[]?>(undefined)`?

File: libs/judicial-system/formatters/src/lib/formatters.ts [hunk 2]
```diff
@@ -12,24 +12,33 @@ const getAsDate = (date: Date | string | undefined | null): Date => {
...
+export function formatDate(
+  date: null,
+  formatPattern: string,
+): string | undefined
 
 export function formatDate(
   date: Date | string | undefined | null,
   formatPattern: string,
-): string | null {
+): string | undefined {
```
Comment (addi) [inline libs/judicial-system/formatters/src/lib/formatters.ts:35]: string?

**Review by addi — CHANGES_REQUESTED** [review]

File: apps/judicial-system/web/src/routes/DetentionRequests/DetentionRequests.tsx (hunk 1 above)
Comment (oddsson) [reply apps/judicial-system/web/src/routes/DetentionRequests/DetentionRequests.tsx:45]: That's not a valid syntax 👇

Screenshot 2020-11-10 at 13 37 22

File: libs/judicial-system/formatters/src/lib/formatters.ts (hunk 2 above)
Comment (oddsson) [reply libs/judicial-system/formatters/src/lib/formatters.ts:35]: All dates are treated as strings because that's how they are sent to the server.

**Review by addi — APPROVED** [review]

**Pull request merged**